src/tools/storage-key-recipe-resolver.ts

```diff
@@ -126,6 +126,10 @@ export class StorageKeyRecipeResolver {
    * @param recipe long-running or ephemeral recipe
    */
   validateHandles(recipe: Recipe) {
+    recipe.handles
+      .filter(h => h.fate === 'create' && !h.capabilities.isTiedToArc && !h.id)
+      .forEach(h => console.warn(`Create handle '${h.localName}' should have an id.`));
```

Comment:
nit: we've agreed to use `for (const h of handles) {` rather than `handles.forEach(h => ...`

Reply:
Is there a place where I can discuss / review this decision? I'd like more context about brightlines.

Reply:
iirc it was in chat a few years back.  the idea is it's more readable and easier to debug, but no actual benefits for the .forEach(...) approach

Reply:
Interesting. I like higher-order functions on arrays in general, though `forEach` is the weakest of these options. I usually use forEach if it's at the end of a chain of other hofs, but I can trade this in for `for...of`. 
One benefit for `for` loops: It makes using async code waay easier.